lib/pbench/server/__init__.py

```diff
@@ -39,24 +45,38 @@ def __init__(self, cfg_name):
 
         # Now fetch some default common pbench settings that are required.
         try:
-            self.TOP = Path(self.conf.get("pbench-server", "pbench-top-dir"))
+            self.TOP = Path(
+                validate_config_val(self.conf.get("pbench-server", "pbench-top-dir"))
```

Comment:
Nice change.

Perhaps we can leverage this new method to reduce the overall "mechanisms" by making `validate_config_val` a method of the `PbenchConfig` base class, and then having it use `self.conf.get` directly.

Something like:

```
def _get_conf(self, section, option):
    """_get_conf - get the option from the section, raising ValueError if it is empty, return the option value as a string."""
    option_val = self.conf.get(section, option)
    if not option_val:
        raise ValueError(f"option {option} in section {section} is empty")
    return option_val
```

And while we're at it, we should DRY this out, adding `_get_valid_dir_option` method to the `PbenchConfig` base class as well:

```
def _get_valid_dir_option(self, section, option):
    """_get_valid_dir_option - get the directory option from the given section, raising a ValueError if it is not a directory, returning a Path directory object."""
    dir_val = self._get_conf(section, option)
    dir_path = Path(dir_val)
    if not dir_path.is_dir():
        raise ValueError(dir_val)
    return dir_path
```

Then we can use it like:

```
try:
    self.TOP = self._get_valid_dir_option("pbench-server", "pbench-tmp-dir")
except ValueError as exc:
    raise BadConfig(f"Bad TOP={exc}")
```